crates/sui-framework/tests/delegation_tests.move

```diff
@@ -21,6 +22,79 @@ module sui::delegation_tests {
+    #[test]
+    fun test_split_join_staked_sui() {
+        let scenario_val = test_scenario::begin(DELEGATOR_ADDR_1);
+        let scenario = &mut scenario_val;
+        // All this is just to generate a dummy StakedSui object to split and join later
+        set_up_sui_system_state(scenario);
+        test_scenario::next_tx(scenario, DELEGATOR_ADDR_1);
+        {
+            let system_state = test_scenario::take_shared<SuiSystemState>(scenario);
+            let system_state_mut_ref = &mut system_state;
+            let ctx = test_scenario::ctx(scenario);
+            sui_system::request_add_delegation(
+                system_state_mut_ref, coin::mint_for_testing(60, ctx), VALIDATOR_ADDR_1, ctx);
+            test_scenario::return_shared(system_state);
+        };
+        // At this moment we have the StakedSui object at the delegator address: send a txn to split it
+        test_scenario::next_tx(scenario, DELEGATOR_ADDR_1);
+        {
+            let staked_sui = test_scenario::take_from_sender<StakedSui>(scenario);
+            let ctx = test_scenario::ctx(scenario);
+            staking_pool::split_staked_sui(&mut staked_sui, 20, ctx);
+            test_scenario::return_to_sender(scenario, staked_sui);
+        };
+
+        // Verify the correctness of the split and merge them back
+        test_scenario::next_tx(scenario, DELEGATOR_ADDR_1);
+        {
+            let staked_sui_ids = test_scenario::ids_for_sender<StakedSui>(scenario);
+            assert!(vector::length(&staked_sui_ids) == 2, 101); // staked sui split to 2 coins
+            
+            let part1 = test_scenario::take_from_sender_by_id<StakedSui>(scenario, *vector::borrow(&staked_sui_ids, 0));
+            let part2 = test_scenario::take_from_sender_by_id<StakedSui>(scenario, *vector::borrow(&staked_sui_ids, 1));
+                
+            let amount1 = staking_pool::staked_sui_amount(&part1);
+            let amount2 = staking_pool::staked_sui_amount(&part2);
+            assert!(amount1 == 20 || amount1 == 40, 102);
+            assert!(amount2 == 20 || amount2 == 40, 103);
+            assert!(amount1 + amount2 == 60, 104);
+
+            staking_pool::join(&mut part1, part2);
+            assert!(staking_pool::staked_sui_amount(&part1) == 60, 105);
+            test_scenario::return_to_sender(scenario, part1);
+        };
+
+        // Create another instance of staked sui with a different epoch activation
+        governance_test_utils::advance_epoch(scenario); // advance epoch
+        test_scenario::next_tx(scenario, DELEGATOR_ADDR_1);
+        {
+            let system_state = test_scenario::take_shared<SuiSystemState>(scenario);
+            let system_state_mut_ref = &mut system_state;
+            let ctx = test_scenario::ctx(scenario);
+            sui_system::request_add_delegation(
+                system_state_mut_ref, coin::mint_for_testing(80, ctx), VALIDATOR_ADDR_1, ctx);
+            test_scenario::return_shared(system_state);
+        };
```

Comment:
Same here -- I believe you can use `governance_utils::delegate_to` here as well if you would like